crates/bevy_tasks/src/task.rs

```diff
@@ -41,6 +41,13 @@ impl<T> Task<T> {
     pub async fn cancel(self) -> Option<T> {
         self.0.cancel().await
     }
+
+    /// Returns `true` if the current task is finished.
+    ///
+    /// Note that in a multithreaded environment, this task can change finish immediately after calling this function.
```

Comment:
I'd like to see a sentence comparing this to polling, and explaining why you might want one vs the other.